#!/usr/bin/env python
PACKAGE = "pr2_cartesian_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_nullspace",   bool_t, 0, "Toggles the usage of nullspace tasks", False)
gen.add("use_kalman_gain",   bool_t, 0, "Toggles the usage of the Kalman gain in the kf estimator", True)
gen.add("vd_amp",   double_t, 0, "Sets the amplitude of the translational velocity signal", 0.0, 0.0, 0.5)
gen.add("vd_freq",   double_t, 0, "Sets the frequency of the translational velocity signal", 0.0, 0.0, 2)
gen.add("wd_amp",   double_t, 0, "Sets the amplitude of the rotational velocity signal", 0.0, 0.0, 1)
gen.add("wd_freq",   double_t, 0, "Sets the frequency of the rotational velocity signal", 0.0, 0.0, 2)
gen.add("force",   double_t, 0, "Sets the target contact force between mechanism parts", 0.0, 0.0, 10)
gen.add("nullspace_gain",   double_t, 0, "Sets the value for the nullspace gain", 0.0, 0.0, 1)
gen.add("constant_observer_gain",   double_t, 0, "Sets the constant gain to use in the linear observer", 0.0, 0.0, 10)

exit(gen.generate(PACKAGE, "mechanism_identification_controller", "MechanismIdentification"))